lib/explorer/series.ex

```diff
@@ -3617,7 +3617,7 @@ defmodule Explorer.Series do
   defp cast_to_pow({:f, l}, {:f, r}), do: {:f, max(l, r)}
   defp cast_to_pow({:f, l}, {n, _}) when K.in(n, [:u, :s]), do: {:f, l}
   defp cast_to_pow({n, _}, {:f, r}) when K.in(n, [:u, :s]), do: {:f, r}
-  defp cast_to_pow({n, _}, {:s, _}) when K.in(n, [:u, :s]), do: {:f, 64}
+  defp cast_to_pow({n, _}, {:s, _}) when K.in(n, [:u, :s]), do: {:s, 64}
```

Comment:
In theory this has to be a flat because the exponent can be negative, which means a float is returned. Did Polars change behaviour here and it now raises for negative exponents?

Reply:
https://github.com/pola-rs/polars/pull/15506

Reply:
If I'm reading that right, we need to disallow raising to a signed int completely now.

Reply:
I think we can still keep the same behaviour - always have a float series if we `pow(any, signed_int)` -, as I did in https://github.com/elixir-explorer/explorer/pull/917/commits/2644d270bf8dfebb22b91c0713a8be9d1c208d10

But I think we could change a little how we create series from scalar numbers. If the number is a positive integer, we could create the series as `:u32` (or adjust to the size that most fits the number). Doing this would use less memory and would remove the surprise of having the result of `pow(signed, signed)` to be a float. WDYT?

Reply:
It is fine to mirror Polars behaviour, because that's also how Nx behaves too. If integers are given (signed or unsigned), it raises for negative values. If you give a float, it preservers the float type.

Reply:
> But I think we could change a little how we create series from scalar numbers.

@philss I think we've had a similar discussion before but I can't find it right now. @lkarthee was also advocating inferring the "simplest" integer dtype from a list of scalars.

IIRC, we decided to not do that because it was limiting in unexpected ways. For example, if we infer `[1, 2, 3]` as `dtype: :u8`, then we can't do subtraction that series since the result may contain negative values.

If I find the discussion I'll update. Also, the possibility is still up for discussion IMO.

Reply:
Update, found it:

* https://github.com/elixir-explorer/explorer/pull/826

Reply:
@billylanchantin thank you for the context! ❤️ 
I had completely forgot about that discussion 😬

Reply:
You betcha!